src/event-bus/EventBus.ts

```diff
@@ -1,4 +1,8 @@
-export type Subscriber = (message: any) => void;
+/**
+ * @param payload The main message that is usually also sent to PutRumEvents
+ * @param internalMessage If payload is a RUM event, then additional information can be communicated here that should not be sent to PutRumEvents
+ */
+export type Subscriber = (payload: any, internalMessage?: any) => void;
```

Comment:
question: Why do we need to pass around this `internalMessage` object? Are there other ways we can solve this problem without modifying the signatures of this and the other functions that use `internalMessage`?

Reply:
I removed the internalMessage interface after dropping our dependency on file extensions to derive fileType.